test/WebJobs.Script.Tests/Helpers/EncryptionHelperTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Helpers
+{
+    public class EncryptionHelperTests : IDisposable
+    {
+        [Fact]
+        public void EncryptShouldThrowIdNoEncryptionKeyDefined()
+        {
+            // Make sure WEBSITE_AUTH_ENCRYPTION_KEY is empty
+            Environment.SetEnvironmentVariable("WEBSITE_AUTH_ENCRYPTION_KEY", string.Empty);
+
+            try
+            {
+                EncryptionHelper.Encrypt("value");
+            }
+            catch (Exception ex)
+            {
+                Assert.IsType<InvalidOperationException>(ex);
+                Assert.Equal("No encryption key defined in the environment.", ex.Message);
+            }
+        }
+
+        [Theory]
+        [InlineData("value")]
+        public void EncryptShouldGenerateDecryptableValues(string valueToEncrypt)
+        {
+            var key = TestHelpers.GenerateKeyBytes();
+            var stringKey = TestHelpers.GenerateKeyHexString(key);
+            Environment.SetEnvironmentVariable("WEBSITE_AUTH_ENCRYPTION_KEY", stringKey);
+
+            var encrypted = EncryptionHelper.Encrypt(valueToEncrypt);
+            var decrypted = EncryptionHelper.Decrypt(key, encrypted);
+
+            Assert.Matches("(.*)[.](.*)[.](.*)", encrypted);
+            Assert.Equal(valueToEncrypt, decrypted);
+        }
+
+        [Fact]
+        public void Encrypt_And_Decrypt_Context_With_Signature()
+        {
+            var websiteAuthEncryptionKey = TestHelpers.GenerateKeyBytes();
+            var websiteAuthEncryptionStringKey = TestHelpers.GenerateKeyHexString(websiteAuthEncryptionKey);
+            var hostContext = GetHostAssignmentContext();
+            var hostContextJson = JsonConvert.SerializeObject(hostContext);
+
+            Environment.SetEnvironmentVariable(EnvironmentSettingNames.ContainerEncryptionKey, string.Empty);
+            Environment.SetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey, websiteAuthEncryptionStringKey);
+
+            var encryptedHostContextWithSignature = EncryptionHelper.Encrypt(hostContextJson, websiteAuthEncryptionKey, includesSignature: true);
```

Comment:
Why not use `TestEnvironment`? The `Encrypt` method has a parameter of type `IEnvironment`.

Reply:
not sure - note these are existing tests. i just renamed file. ill leave as is for now to demonstrate tests run as before with these changes

Reply:
agree they could be improved in future